aiidalab_widgets_base/elns.py

```diff
@@ -0,0 +1,361 @@
+            return
+
+        traitlets.dlink((eln, "node"), (self, "node"))
+        eln.import_data_object()
+
+
+class ElnExportWidget(ipw.VBox):
+    node = traitlets.Instance(Node, allow_none=True)
+
+    def __init__(self, **kwargs):
+
+        # Send to ELN button.
+        self.send = ipw.Button(description="Send to ELN")
+        self.send.on_click(self.send_to_eln)
+
+        # Use non-default destination.
+        self.modify_settings = ipw.Checkbox(
+            description="Update destination.", indent=False
+        )
+        self.modify_settings.observe(self.handle_output, "value")
+
+        # Used to output additional settings.
+        self._output = ipw.Output()
+
+        # Communicate to the user if something isn't right.
+        self.error_message = ipw.HTML()
+
+        children = [
+            ipw.HBox([self.send, self.modify_settings]),
+            self.error_message,
+            self._output,
+        ]
+        self.eln = connect_to_eln()
+
+        super().__init__(children=children, **kwargs)
+
+    @traitlets.observe("node")
+    def _observe_node(self, _=None):
+        if self.node is None:
+            return
+
+        if "eln" in self.node.extras:
+            info = self.node.extras["eln"]
+        else:
+            try:
+                q = QueryBuilder().append(
+                    Node,
+                    filters={"extras": {"has_key": "eln"}},
+                    tag="source_node",
+                    project="extras.eln",
+                )
+                q.append(
+                    Node, filters={"uuid": self.node.uuid}, with_ancestors="source_node"
+                )
+                info = q.all(flat=True)[0]
+            except IndexError:
+                info = {}
+
+        self.eln.set_sample_config(**info)
+
+    def send_to_eln(self, _=None):
+        if self.eln.is_connected:
+            self.error_message.value = ""
+            self.eln.export_data_object(data_object=self.node)
+        else:
+            self.error_message.value = """Warning! The access to ELN is not configured. Please follow <a href="https://aiidalab-demo.materialscloud.org/user/aliaksandr.yakutovich@epfl.ch/apps/apps/aiidalab-widgets-base/eln_configure.ipynb" target="_blank">the link</a> to configure it."""
+
+    def handle_output(self, _=None):
+        with self._output:
+            clear_output()
+            if self.modify_settings.value:
+                display(self.eln.sample_config_editor())
+
+
+class ElnConfigureWidget(ipw.VBox):
+    selected = traitlets.Dict()
+
+    def __init__(self, **kwargs):
+        self._output = ipw.Output()
+
+        self.eln_instance = ipw.Dropdown(
+            description="ELN:",
+            options=("Setup new ELN", {}),
```

Comment:
```suggestion
            options=("Set up new ELN", {}),
```

Reply:
as this is currently an option, clicking on "set as default" will set this as default. 
What I would maybe add is a check that this key is in the dict with url and token before it is set as default